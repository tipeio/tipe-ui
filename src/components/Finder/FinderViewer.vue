<template>
  <div
    :data-tipe-ui="$options.name"
    class="viewer"
  >
    <div class="scrollable">
      <div :class="layoutClass">
        <type-file-icon
          v-for="(file, i) in files"
          :key="i"
          :file="file"
          class="item"
        />
      </div>
    </div>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import FileType from './FileType'

import TypeFileIcon from './FileIcon.vue'

export default {
  name: 'TipeFinder',
  components: {
    TypeFileIcon
  },
  props: {
    files: vueTypes.arrayOf(FileType).def([]),
    layout: vueTypes.oneOf(['list', 'grid']).def('list')
  },
  computed: {
    layoutClass() {
      return {
        layout: true,
        [this.layout]: true
      }
    }
  }
}
</script>

<style lang="postcss" scoped>
.viewer {
  overflow: hidden;
}

/* hide scrollbar */
.scrollable {
  width: 100%;
  height: 100%;
  overflow-y: scroll;
  padding-right: 17px;
  box-sizing: content-box;
}

.layout.list {
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: max-content;
  grid-gap: 0.8rem;
}

.layout.grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(11rem, 1fr));
  grid-auto-rows: max-content;
  grid-gap: 0.8rem;
}
</style>
