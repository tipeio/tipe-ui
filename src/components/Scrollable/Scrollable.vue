<template>
  <div :class="$style.window">
    <div :class="$style.scrollable">
      <slot/>
    </div>
  </div>
</template>

<style lang="postcss" module>
.window {
  height: 100%;
  width: 100%;
  position: relative;
  overflow: hidden;
}

.scrollable {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  overflow-y: scroll;
  padding-right: 17px;
  box-sizing: content-box;
}
</style>
