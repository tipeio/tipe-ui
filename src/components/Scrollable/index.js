export { default } from './Scrollable.vue'
