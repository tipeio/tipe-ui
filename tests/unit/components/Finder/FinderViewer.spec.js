import { shallowMount, mount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import FinderViewer from '@/components/Finder/FinderViewer'
import FinderIcon from '@/components/Finder/FileIcon'
import Scrollable from '@/components/Scrollable'
import { mocks } from '@tipe/tipe-test-utils'

describe('FinderViewer', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(FinderViewer)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(FinderViewer)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeFinder')
    })
  })

  describe(':props', () => {
    it(':file - should pass file to FinderIcon component', () => {
      const stubs = {
        TipeScrollable: Scrollable,
        TipeFileIcon: '<!---->'
      }
      const propsData = { files: [mocks.file()] }
      const wrapper = mount(FinderViewer, { propsData, stubs })

      expect(wrapper.find(FinderIcon).props()).toMatchObject(propsData.files[0])
    })

    it(':layout(grid) - should pass layout prop to layout class', () => {
      const stubs = {
        TipeScrollable: Scrollable,
        TipeFileIcon: '<!---->'
      }
      const propsData = { layout: 'grid' }
      const wrapper = mount(FinderViewer, { propsData, stubs })

      expect(wrapper.find('.layout').classes()).toContain('grid-rectangle')
    })
  })
})
