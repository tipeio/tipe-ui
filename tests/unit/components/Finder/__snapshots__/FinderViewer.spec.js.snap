// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinderViewer <template> matches previous snapshot 1`] = `
<div data-tipe-ui="TipeFinder">
  <TipeScrollable-stub></TipeScrollable-stub>
</div>
`;
